context.go

```diff
@@ -16,14 +16,41 @@ import (
 	"github.com/sdboyer/gps"
 )
 
+type BuildContext interface {
+	SourceManager() (gps.SourceManager, error)
+
+	// LoadProject takes a path and searches up the directory tree for a project
+	// root. If an absolute path is given, the search begins in that directory. If
+	// a relative or empty path is given, the search start is computed from the
+	// current working directory. The search stops when a file with the name
+	// ManifestName (manifest.json, by default) is located.
+	//
+	// The Project contains the parsed manifest as well as a parsed lock file, if
+	// present. The import path is calculated as the remaining path segment below
+	// Ctx.GOPATH/src.
+	LoadProject(path string) (*Project, error)
+
+	// SplitAbsoluteProjectRoot takes an absolute path and compares it against
+	// declared GOPATH(s) to determine what portion of the input path should be
+	// treated as an import path - as a project root.
+	//
+	// The second returned string indicates which GOPATH value was used.
+	SplitAbsoluteProjectRoot(path string) (string, error)
+
+	VersionInWorkspace(root gps.ProjectRoot) (gps.Version, error)
+
+	// Returns the root directory where packages can be found.
+	Root() string
```

Comment:
Eh...I could be convinced of one for facilitating searching for directories, but I don't think having this `Root()` method adds much value. We don't want to rename this away from GOPATH, because that's what people already think of wrt "a space on disk where your source packages are kept"; a new name would just add confusion.